modules/snapshot/main.tf

```diff
@@ -60,3 +60,14 @@ resource "aws_lambda_permission" "this" {
   function_name       = trimprefix(data.aws_arn.function.resource, "function:")
   source_arn          = aws_cloudwatch_event_rule.trigger.arn
 }
+
+resource "aws_lambda_invocation" "snapshot" {
+  // initial snapshot invocation
+  function_name = data.aws_arn.function.arn
```

Comment:
Shouldn't this be the function name and not the ARN?

Reply:
I couldnt find the function object for the function_name, it seems to like the full arn.

Reply:
>

Reply:
> it seems to like the full arn

A `terraform plan` adds very little beyond `terraform validate`. It does refresh all your state which requires valid credentials. But it doesn't really do anything to prove that your proposed configuration for a new resource works.

The API take a function name so this is definitely going to result in a 4xx error when `terraform apply` runs:

https://docs.aws.amazon.com/lambda/latest/dg/API_Invoke.html#API_Invoke_RequestSyntax

> I couldnt find the function object

It's this:

https://github.com/observeinc/terraform-aws-lambda/blob/719216702da9216906b66da0f0cb2b31c3e950cf/modules/snapshot/variables.tf#L4-L7

Add `function_name` to that and it'll come through. Objects are to data in Terraform as interfaces are to method receivers in Go. They "select" matching fields and pass through a subset of the value with only the declared schema, stripping out any other keys and doing type coercion/validation as needed.

Reply:
done.